src/MySql/custom/New-AzMySqlFlexibleServer.ps1

```diff
@@ -147,60 +183,418 @@ function New-AzMySqlFlexibleServer {
 
     process {
         try {
-            if ($PSBoundParameters.ContainsKey('Name')) {
-              $PSBoundParameters.ServerName = $PSBoundParameters['Name']
-              $null = $PSBoundParameters.Remove('Name')
+            If (!$PSBoundParameters.ContainsKey('Location')) {
+                $PSBoundParameters.Location = 'westus2'
+            }
+
+            If (!$PSBoundParameters.ContainsKey('AdministratorLoginPassword')) {
+                $Password = Get-GeneratePassword
+                $PSBoundParameters.AdministratorLoginPassword = $Password | ConvertTo-SecureString -AsPlainText -Force
+            }
+            $PSBoundParameters.AdministratorLoginPassword = . "$PSScriptRoot/../utils/Unprotect-SecureString.ps1" $PSBoundParameters['AdministratorLoginPassword']
+
+            If (!(Get-Module -ListAvailable -Name Az.Resources)) { Throw 'Please install Az.Resources module by entering "Install-Module -Name Az.Resources"'}
+            Else { Import-Module -Name Az.Resources }
+            If(!$PSBoundParameters.ContainsKey('ResourceGroupName')) {
+                $PSBoundParameters.ResourceGroupName = Get-RandomNumbers -Prefix 'group' -Length 10
+                $Msg = "Creating Resource Group {0}..." -f $PSBoundParameters.ResourceGroupName
+                Write-Host $Msg
+                $null = New-AzResourceGroup -Name $PSBoundParameters.ResourceGroupName -Location $PSBoundParameters.Location
```

Comment:
Use should process before create a resource group and use $Force parameter to skip the should process judgement
https://docs.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-shouldprocess?view=powershell-7.1

Reply:
Added!